WordPressAuthenticator/Signin/LoginPrologueViewController.swift

```diff
@@ -199,32 +199,73 @@ class LoginPrologueViewController: LoginViewController {
+
+        setButtonViewMargins(forWidth: view.frame.width)
+
+        buttonViewController.setupTopButton(title: loginTitle, isPrimary: true, accessibilityIdentifier: "Prologue Continue Button", onTap: loginTapCallback())
 
+        if configuration.enableUnifiedAuth {
+            buttonViewController.setupBottomButton(title: siteAddressTitle, isPrimary: false, accessibilityIdentifier: "Prologue Self Hosted Button", onTap: siteAddressTapCallback()) 
+        }
+
+        showCancelIfNeccessary(buttonViewController)
+
+        setButtonViewControllerBackground(buttonViewController)
+    }
+    
+    private func buildUnifiedPrologueButtonsWithSiteAddressFirst(_ buttonViewController: NUXButtonViewController, loginTitle: String, siteAddressTitle: String) {
+        guard configuration.enableUnifiedAuth == true else {
+            return
+        }
+        
         setButtonViewMargins(forWidth: view.frame.width)
+
+        buttonViewController.setupTopButton(title: siteAddressTitle, isPrimary: true, accessibilityIdentifier: "Prologue Self Hosted Button", onTap: siteAddressTapCallback())
+        
+        buttonViewController.setupBottomButton(title: loginTitle, isPrimary: false, accessibilityIdentifier: "Prologue Continue Button", onTap:loginTapCallback())
         
-        buttonViewController.setupTopButton(title: loginTitle, isPrimary: true, accessibilityIdentifier: "Prologue Continue Button") { [weak self] in
+        showCancelIfNeccessary(buttonViewController)
+
+        setButtonViewControllerBackground(buttonViewController)
+    }
+    
+    private func siteAddressTapCallback() -> NUXButtonViewController.CallBackType {
+        return { [weak self] in
+            self?.siteAddressTapped()
+        }
+    }
+
+    private func loginTapCallback() -> NUXButtonViewController.CallBackType {
+        return { [weak self] in
             guard let self = self else {
                 return
             }
             
             self.tracker.track(click: .continueWithWordPressCom)
             self.continueWithDotCom()
         }
-
-        if configuration.enableUnifiedAuth {
-            buttonViewController.setupBottomButton(title: siteAddressTitle, isPrimary: false, accessibilityIdentifier: "Prologue Self Hosted Button") { [weak self] in
-                self?.siteAddressTapped()
-            }
-        }
-
+    }
+    
+    private func showCancelIfNeccessary(_ buttonViewController: NUXButtonViewController) {
         if showCancel {
             let cancelTitle = NSLocalizedString("Cancel", comment: "Button title. Tapping it cancels the login flow.")
             buttonViewController.setupTertiaryButton(title: cancelTitle, isPrimary: false) { [weak self] in
                 self?.dismiss(animated: true, completion: nil)
             }
         }
-
-        // Set the button background color to clear so the blur effect blurs the Prologue background color.
```

Comment:
I would leave this comment inside of `setButtonViewControllerBackground`. It's not apparent why the bg color has to be set to clear.

Reply:
That's a good point, that's for Ponting it out. I will merge develop into this branch, solve the conflicts, add the comment back, and merge.